src/core/reducers/categories.js

```diff
@@ -96,6 +104,47 @@ export function createEmptyCategoryList(): EmptyCategoryListType {
   }, {});
 }
 
+export const getCategories = (
+  categoriesState: CategoryMapType,
+  appName: string,
+  addonType: string,
+): Array<mixed> | null => {
+  if (
+    categoriesState &&
+    categoriesState[appName] &&
+    categoriesState[appName][addonType]
+  ) {
+    return Object.values(categoriesState[appName][addonType]);
+  }
+  return null;
+};
+
+export function getCategoryNames(
+  categoriesState: CategoryMapType,
+  addonCategories: AddonCategories,
+  appName: string,
+  addonType: string,
+): Array<mixed> | null {
+  invariant(categoriesState, 'categories state can not be empty!');
```

Comment:
The exclamation point at the end is unnecessary. Same goes for the other `invariant` calls.

Reply:
Ohk! I will remove the exclamations.